templates/stamp/utilities.sh

```diff
@@ -1062,10 +1072,39 @@ start_haproxy()
 {
     log "Starting HA Proxy Server"
 
-    service haproxy start
+    # add some resilience
+    local mysql_admin_username="${1}"
+    local mysql_admin_password="${2}"
+    local haproxy_server="${3}"
+    local haproxy_port="${4}"
+
+    local server_started=0
+    local wait_time_seconds=10
+    local max_wait_seconds=$(($wait_time_seconds * 10))
+    local total_wait_seconds=0
+
+    while [[ $server_started == 0 ]] ;
+    do
+        service haproxy start
+
+        # run a basic query against the Proxy
+        db_response=`mysql -u ${mysql_admin_username} -p${mysql_admin_password} -h ${haproxy_server} -P ${haproxy_port} -e "SHOW DATABASES;"`
 
-    # Important - wait for the service to startup and interact with mysql
-    sleep 15s
+        if [[ -z "${db_response// }" ]];
```

Comment:
nit: please add a comment above this line explaining what ${var// } does

Reply:
- [ ] add comments about use of ${var// }